plugins/auto-sizes/includes/improve-calculate-sizes.php

```diff
@@ -288,15 +296,43 @@ function auto_sizes_filter_render_block_context( array $context, array $block ):
 	);
 
 	if ( in_array( $block['blockName'], $provider_blocks, true ) ) {
-		$alignment = $block['attrs']['align'] ?? '';
+		// Normalize default alignment values.
+		$alignment = isset( $block['attrs']['align'] ) && '' !== $block['attrs']['align'] ? $block['attrs']['align'] : 'default';
+		// Use the defined constant for constraints.
+		$constraints = AUTO_SIZES_CONSTRAINTS;
 
-		// If the container block doesn't have alignment, it's assumed to be 'default'.
-		if ( '' === $alignment ) {
-			$context['max_alignment'] = 'default';
-		} elseif ( 'wide' === $alignment ) {
-			$context['max_alignment'] = 'wide';
+		if ( 'default' === $alignment || 'wide' === $alignment ) {
+			$context['max_alignment'] = $constraints[ $context['max_alignment'] ] > $constraints[ $alignment ] ? $context['max_alignment'] : $alignment;
 		}
 	}
 
+	if ( 'core/columns' === $block['blockName'] ) {
+		// This is a special context key just to pass to the child 'core/column' block.
+		$context['column_count'] = count( $block['innerBlocks'] );
+	}
+
+	if ( 'core/column' === $block['blockName'] ) {
+		$found_image_block = wp_get_first_block( $block['innerBlocks'], 'core/image' );
+		$found_cover_block = wp_get_first_block( $block['innerBlocks'], 'core/cover' );
+		if ( count( $found_image_block ) > 0 || count( $found_cover_block ) > 0 ) {
+			// Get column width, if explicitly set.
+			if ( isset( $block['attrs']['width'] ) && '' !== $block['attrs']['width'] ) {
+				$current_width = floatval( rtrim( $block['attrs']['width'], '%' ) ) / 100;
+			} elseif ( isset( $parent_block->context['column_count'] ) && $parent_block->context['column_count'] ) {
+				// Default to equally divided width if not explicitly set.
+				$current_width = 1 / $parent_block->context['column_count'];
+			} else {
+				// Full width fallback.
+				$current_width = 1;
```

Comment:
Nit-pick: Use floats to clarify that this value should generally be a floating point number.
```suggestion
				$current_width = 1.0 / $parent_block->context['column_count'];
			} else {
				// Full width fallback.
				$current_width = 1.0;
```

Reply:
Addressed in https://github.com/WordPress/performance/pull/1795/commits/d74494e4d152d42b1eff6103a70b9b0ea2181277